### ISSUE TITLE
Manuscript about Best Practices development

### ISSUE THREAD

**Issue by bgruening** [galaxy-iuc/standards#18]
What does the IUC thinks about writing a manuscript about Best Practices of tool development. 
We all have seen many bad habits during our tool wrapping work and we gained a lot of experience. I really think this is worth to write down.

My idea is to write about the following ideas:
- [X things you should take into consideration if you code for big data](https://docs.google.com/document/d/1P2A5M5pI1ahZJbBOK2TeIrSYMPgzxwHmsrG3iENszKE)
- Writing about tools that will run on cluster environments, headless ... (not strictly bound to Galaxy)
- The necessity to have a best practice guide and code reviews including automated testing
- A more Galaxy specific section about the IUC and this repository (+planemo and additional community tools to assist in tool development)

Opinions? Ideas?

**Comment by hexylena** [comment]
I'd be in favour of that. Just compiling all of the knowledge we have scattered across PRs and the wiki and here. Also things like the planemo "How do I ... ?" would likely be useful

**Comment by bgruening** [comment]
Thanks @erasche I updated the original comment, feel free to do this as well.

**Comment by peterjc** [comment]
When you say manuscript, do you mean formal journal publication? We could ping Scott Edmunds to see if it would fit the Galaxy series at GigaScience: http://www.gigasciencejournal.com/series/Galaxy

**Comment by ScottBGI** [comment]
It most certainly would fit, and would also be very useful for us to then link to it from our author instructions for future Galaxy submissions. To avoid duplication you should consult with the GCC data wranglers as they are putting together a best practices manuscript too, but it sounds like this would be more on the steps before their work.

**Comment by bgruening** [comment]
@peterjc yes a formal manuscript. I think this can be also interesting for non-Galaxy developers.
Thanks @ScottBGI for commenting!

**Comment by blankenberg** [comment]
I am +1 for this. I like the idea of a general primer for writing tools for big data and then showing how following these generic best practices are needed to enable smooth incorporation into analysis pipelines (including wrapping for inclusion in Galaxy as a specific use case).

**Comment by bgruening** [comment]
@yhoogstrate here it is!

@galaxy-iuc/iuc has anyone time to get this started?
